lib/buffer.js

```diff
@@ -337,12 +337,18 @@ function slowToString(encoding, start, end) {
   }
 }
 
-
 Buffer.prototype.toString = function() {
   const length = this.length | 0;
-  if (arguments.length === 0)
-    return this.utf8Slice(0, length);
-  return slowToString.apply(this, arguments);
+  if (arguments.length === 0) {
+    var result = this.utf8Slice(0, length);
+    if (result === undefined)
+      throw new Error('toString failed');
+    return result;
+  }
+  var result = slowToString.apply(this, arguments);
+  if (result === undefined)
+    throw new Error('toString failed');
+  return result;
```

Comment:
Can be made a little more compact if you write it like this:

```javascript
if (arguments.length === 0) {
  var result = this.utf8Slice(0, this.length);
} else {
  var result = slowToString.apply(this, arguments);
}
if (result === undefined)
  throw new Error('toString failed');
return result;
```

Can you the blank line at the top back in?